jquery-ajax-localstorage-cache.js

```diff
@@ -7,6 +7,13 @@
      * or one generated from the url, the type and, if present, the data.
      */
     var genCacheKey = function (options) {
+        //in case cachey key is required to be generated dynamically by a function
+        if(
+            options.cacheKey && 
+            typeof(options.cacheKey) == typeof(function(){})
+        ) {
+            return options.cacheKey(options);
```

Comment:
My main concern with allowing dynamic generation of cacheKey's is that the cacheKey must remain stable between calls from ajaxPrefilter and ajaxTransport. This can be made the developer's concern, but should then be noted in the ReadMe that the passed generation function must always return the same cacheKey if given the same parameters.